### ISSUE TITLE
R crashes on data.table::fread with space-delimited file and verbose=TRUE

### ISSUE THREAD

**Issue by chrisvwn** [Rdatatable/data.table#5643]
#### Thanks! Please remove the text above and include the two items below.

`#` [`Minimal reproducible example`](https://stackoverflow.com/questions/5963269/how-to-make-a-great-r-reproducible-example); please be sure to set `verbose=TRUE` where possible!

```
  nrow_read <- 1200000
  nrow_skip <- 0
  nrow_df <- nrow_read
  log_fname = "sweep_20230513_1100.log" #attached

  swp_in <- data.table::fread(input = log_fname,
                                header = FALSE,
                                skip = nrow_skip,
                                nrow = nrow_read,
                                verbose = TRUE)
```

`#` `Output of sessionInfo()`
```
R version 4.2.3 (2023-03-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: DragonOS FocalX R27.1

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8    LC_NUMERIC=C            LC_TIME=sw_KE           LC_COLLATE=en_US.UTF-8 
 [5] LC_MONETARY=sw_KE       LC_MESSAGES=en_US.UTF-8 LC_PAPER=sw_KE          LC_NAME=C              
 [9] LC_ADDRESS=C            LC_TELEPHONE=C          LC_MEASUREMENT=sw_KE    LC_IDENTIFICATION=C    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bit64_4.0.5       bit_4.0.5         data.table_1.14.8 dplyr_1.1.1      

loaded via a namespace (and not attached):
 [1] fansi_1.0.4      utf8_1.2.3       R6_2.5.1         lifecycle_1.0.3  magrittr_2.0.3  
 [6] pillar_1.9.0     rlang_1.1.0      cli_3.6.1        rstudioapi_0.14  vctrs_0.6.1     
[11] generics_0.1.3   tools_4.2.3      serial_3.0       glue_1.6.2       compiler_4.2.3  
[16] pkgconfig_2.0.3  tcltk_4.2.3      tidyselect_1.2.0 tibble_3.2.1 
```
[sweep_20230513_1100.log](https://github.com/Rdatatable/data.table/files/11558373/sweep_20230513_1100.log)

**Comment by ben-schwen** [comment]
Ty for the report. Your example works on the current dev version. You can update to dev version via `data.table::update_dev_pkg()`

**Comment by MiguelAngelTorres** [comment]
Is this issue solved? Should be closed?

**Comment by ben-schwen** [comment]
@MiguelAngelTorres what was missing is a git bisect to know when the issue was fixed for future reference.

Just did a git bisect and this was solved in #5046.
